ql/instruments/basketoption.hpp

```diff
@@ -36,7 +36,7 @@ namespace QuantLib {
       private:
         boost::shared_ptr<Payoff> basePayoff_;
       public:
-        BasketPayoff(const boost::shared_ptr<Payoff> &p)
+        explicit BasketPayoff(const boost::shared_ptr<Payoff> &p)
```

Comment:
Technically, removing an implicit conversion would break backward compatibility.  But in this case I think it's ok. If some code used this, it probably deserves to be broken...